server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -345,6 +371,27 @@ static ClusterState updateIndexMetadata(String subscriptionName,
         }
     }
 
+    /**
+     * @return empty list if there is no validation error and all tables don't exist locally
+     * or list of existing tables otherwise.
+     * These tables will be reported
+     * and will be visible in pg_subscriptions_rel table.
+     */
+    private static List<RelationName> getExistingLocallyTables(Subscription subscription,
+                                                               ClusterState subscriberClusterState,
+                                                               Collection<RelationName> updatedRelationsList) {
+        // Table existence check is done on a subscription creation but we need such check on
+        // every remote state update to cover case when several clusters' subscription chain forms a loop,
+        // no tables at the beginning and one of the cluster subscribed to ALL TABLES.
+        // In such case the cluster can transitively subscribe to its own table.
+        var metadata = subscriberClusterState.metadata();
+        Set currentlyReplicatedTables = subscription.relations().keySet();
```

Comment:
```suggestion
        Set<RelationName> currentlyReplicatedTables = subscription.relations().keySet();
```

No need for raw set?